crates/anodized-core/src/test_instrument_fn.rs

```diff
@@ -444,3 +444,40 @@ fn test_instrument_with_complex_mixed_conditions() {
     let observed = instrument_fn_body(&spec, &body, is_async).unwrap();
     assert_block_eq(&observed, &expected);
 }
+
+#[test]
+fn test_instrument_with_clones() {
+    let spec: Spec = parse_quote! {
+        requires: CONDITION_1,
+        clones: [
+            EXPR_1 as ALIAS_1,
+            EXPR_2 as ALIAS_2,
+        ],
+        ensures: [
+            CONDITION_2,
+            CONDITION_3,
+        ],
+    };
+    let body = make_fn_body();
+    let is_async = false;
+
+    let expected: Block = parse_quote! {
+        {
+            assert!(CONDITION_1, "Precondition failed: CONDITION_1");
+            let (ALIAS_1, ALIAS_2) = ((EXPR_1).clone(), (EXPR_2).clone());
+            let __anodized_output = #body;
```

Comment:
When the body runs, it now has a bunch of new names in scope. Not ideal. Seems possible to fix by hoisting the body and defining it at the top as a closure.